### PR TITLE
Migrate adev to use the control flow syntax 

### PR DESCRIPTION
**Pull request by JeanMeche**
Note: this commit does not update the first-app tutorial to the control flow as the video still reference `*ngFor`

### CODE REVIEW

File: adev/src/app/editor/terminal/terminal.component.ts [hunk 1]
```diff
@@ -23,15 +23,14 @@ import {debounceTime} from 'rxjs/operators';
...
-import {NgIf} from '@angular/common';
 
 @Component({
   selector: 'docs-tutorial-terminal',
   standalone: true,
   templateUrl: './terminal.component.html',
   styleUrls: ['./terminal.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
-  imports: [NgIf],
+  imports: [],
```
Comment (twerske) [inline adev/src/app/editor/terminal/terminal.component.ts:33]: nit: removable

**Review by twerske — APPROVED** [review]
A few nits on the else blocks and empty imports that the script didn't catch, but those can be follow up PRs

File: adev/src/content/api-examples/common/pipes/ts/keyvalue_pipe.ts [hunk 2]
```diff
@@ -14,11 +14,15 @@ import {Component} from '@angular/core';
   template: `
     <span>
       <p>Object</p>
-      <div *ngFor="let item of object | keyvalue">{{ item.key }}:{{ item.value }}</div>
+      @for (item of object | keyvalue; track item) {
```
Comment (otodockal) [inline adev/src/content/api-examples/common/pipes/ts/keyvalue_pipe.ts:17]:
```suggestion
      @for (item of object | keyvalue; track item.key) {
```

File: adev/src/content/api-examples/common/pipes/ts/keyvalue_pipe.ts [hunk 3]
```diff
@@ -14,11 +14,15 @@ import {Component} from '@angular/core';
...
+        <div>{{ item.key }}:{{ item.value }}</div>
+      }
       <p>Map</p>
-      <div *ngFor="let item of map | keyvalue">{{ item.key }}:{{ item.value }}</div>
+      @for (item of map | keyvalue; track item) {
```
Comment (otodockal) [inline adev/src/content/api-examples/common/pipes/ts/keyvalue_pipe.ts:21]:
```suggestion
      @for (item of map | keyvalue; track item.key) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 4]
```diff
@@ -7,82 +7,82 @@ <h1 id="toc">Template Syntax</h1>
...
+        <div class="context">
+          {{ title }}
+          <span [hidden]="isUnchanged">changed</span>
+        </div>
+
+
+        <p>Template input variable expression context (let hero)</p>
+        <!-- template hides the following; plenty of examples later -->
+        <ng-template>
+          @for (hero of heroes; track hero) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:72]:
```suggestion
          @for (hero of heroes; track hero.id) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 5]
```diff
@@ -191,563 +191,565 @@ <h3>
...
+                          <div [style.display]="isSpecial ? 'block' : 'none'">Show with style</div>
+                          <div [style.display]="isSpecial ? 'none'  : 'block'">Hide with style</div>
+
+                          <a class="to-toc" href="#toc">top</a>
+
+                          <!-- NgFor binding -->
+                          <hr><h2 id="ngFor">NgFor Binding</h2>
+
+                          <div class="box">
+                            @for (hero of heroes; track hero) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:487]:
```suggestion
                            @for (hero of heroes; track hero.id) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 6]
```diff
@@ -191,563 +191,565 @@ <h3>
...
+                              <div>{{ hero.name }}</div>
+                            }
+                          </div>
+                          <br>
+
+                            <div class="box">
+                              <!-- *ngFor w/ hero-detail Component -->
+                              @for (hero of heroes; track hero) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:495]:
```suggestion
                              @for (hero of heroes; track hero.id) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 7]
```diff
@@ -191,563 +191,565 @@ <h3>
...
+                                <app-hero-detail [hero]="hero"></app-hero-detail>
+                              }
+                            </div>
+
+                            <a class="to-toc" href="#toc">top</a>
+
+                            <h4 id="ngFor-index">*ngFor with index</h4>
+                            <p>with <em>semi-colon</em> separator</p>
+                            <div class="box">
+                              @for (hero of heroes; track hero; let i = $index) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:505]:
```suggestion
                              @for (hero of heroes; track hero.id; let i = $index) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 8]
```diff
@@ -191,563 +191,565 @@ <h3>
...
+                                <div>{{ i + 1 }} - {{ hero.name }}</div>
+                              }
+                            </div>
+
+                            <p>with <em>comma</em> separator</p>
+                            <div class="box">
+                              <!-- Ex: "1 - Hercules" -->
+                              @for (hero of heroes; track hero; let i = $index) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:513]:
```suggestion
                              @for (hero of heroes; track hero.id; let i = $index) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 9]
```diff
@@ -191,563 +191,565 @@ <h3>
...
+                            <a class="to-toc" href="#toc">top</a>
+
+                            <h4 id="ngFor-trackBy">*ngFor trackBy</h4>
+                            <button type="button" (click)="resetHeroes()">Reset heroes</button>
+                            <button type="button" (click)="changeIds()">Change ids</button>
+                            <button type="button" (click)="clearTrackByCounts()">Clear counts</button>
+
+                            <p><em>without</em> trackBy</p>
+                            <div class="box">
+                              @for (hero of heroes; track hero) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:527]:
```suggestion
                              @for (hero of heroes; track hero.id) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 10]
```diff
@@ -191,563 +191,565 @@ <h3>
...
+                              @if (heroesNoTrackByCount) {
+                                <div id="noTrackByCnt" >
+                                  Hero DOM elements change #{{ heroesNoTrackByCount }} without trackBy
+                                </div>
+                              }
+                            </div>
+
+                            <p>with trackBy</p>
+                            <div class="box">
+                              @for (hero of heroes; track trackByHeroes($index, hero)) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:540]:
```suggestion
                              @for (hero of heroes; track hero.id) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 11]
```diff
@@ -191,563 +191,565 @@ <h3>
...
+                                  Hero DOM elements change #{{heroesWithTrackByCount}} with trackBy
+                                </div>
+                              }
+                            </div>
+
+                            <br><br><br>
+
+                            <p>with trackBy and <em>semi-colon</em> separator</p>
+                            <div class="box">
+                              @for (hero of heroes; track trackByHeroes($index, hero)) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:555]:
```suggestion
                              @for (hero of heroes; track hero.id) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 12]
```diff
@@ -191,563 +191,565 @@ <h3>
...
+                                <div>({{ hero.id }}) {{ hero.name }}</div>
+                              }
+                            </div>
+
+                            <p>with trackBy and <em>comma</em> separator</p>
+                            <div class="box">
+                              @for (hero of heroes; track hero) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:562]:
```suggestion
                              @for (hero of heroes; track hero.id) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 13]
```diff
@@ -191,563 +191,565 @@ <h3>
...
+                                <div>({{ hero.id }}) {{ hero.name }}</div>
+                              }
+                            </div>
+
+                            <p>with trackBy and <em>space</em> separator</p>
+                            <div class="box">
+                              @for (hero of heroes; track hero) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:569]:
```suggestion
                              @for (hero of heroes; track hero.id) {
```

File: adev/src/content/examples/template-syntax/src/app/app.component.html [hunk 14]
```diff
@@ -191,563 +191,565 @@ <h3>
...
+                                <div>({{ hero.id }}) {{ hero.name }}</div>
+                              }
+                            </div>
+
+                            <p>with <em>generic</em> trackById function</p>
+                            <div class="box">
+                              @for (hero of heroes; track hero) {
```
Comment (otodockal) [inline adev/src/content/examples/template-syntax/src/app/app.component.html:576]:
```suggestion
                              @for (hero of heroes; track hero.id) {
```

File: adev/src/content/examples/testing/src/app/dashboard/dashboard.component.html [hunk 15]
```diff
@@ -2,7 +2,7 @@ <h2 highlight>{{ title }}</h2>
 
 <div class="grid grid-pad">
   <!-- #docregion dashboard-hero -->
-  @for (hero of heroes; track hero) {
+  @for (hero of heroes; track $index) {
```
Comment (otodockal) [inline adev/src/content/examples/testing/src/app/dashboard/dashboard.component.html:5]:
```suggestion
  @for (hero of heroes; track hero.id) {
```

File: adev/src/content/api-examples/forms/ts/reactiveSelectControl/reactive_select_control_example.ts [hunk 16]
```diff
@@ -15,15 +15,17 @@ import {FormControl, FormGroup} from '@angular/forms';
   template: `
     <form [formGroup]="form">
       <select formControlName="state">
-        <option *ngFor="let state of states" [ngValue]="state">
-          {{ state.abbrev }}
-        </option>
+        @for (state of states; track state) {
```
[the remark this answers is not in the document]
Comment (JeanMeche) [reply adev/src/content/api-examples/forms/ts/reactiveSelectControl/reactive_select_control_example.ts:18]: let's use `$index`, it's a reasonable choice here.

File: adev/src/content/api-examples/forms/ts/selectControl/select_control_example.ts [hunk 17]
```diff
@@ -15,15 +15,17 @@ import {Component} from '@angular/core';
     <form #f="ngForm">
       <select name="state" ngModel>
         <option value="" disabled>Choose a state</option>
-        <option *ngFor="let state of states" [ngValue]="state">
-          {{ state.abbrev }}
-        </option>
+        @for (state of states; track state) {
```
[the remark this answers is not in the document]
Comment (JeanMeche) [reply adev/src/content/api-examples/forms/ts/selectControl/select_control_example.ts:18]: let's use `$index`, it's a reasonable choice here.

**Pull request closed without merging**